Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -68,20 +67,17 @@ class ClipboardBarDisplayHandler: NSObject, URLChangeDelegate {
         firstTab.observeURLChanges(delegate: self)
     }
 
-    @objc private func didRestoreSession() {
-        DispatchQueue.main.sync {
-            if let tabManager = self.tabManager,
-                let firstTab = tabManager.selectedTab {
-                self.observeURLForFirstTab(firstTab: firstTab)
-            } else {
-                firstTabLoaded = true
-            }
-
-            NotificationCenter.default.removeObserver(self, name: .DidRestoreSession, object: nil)
-
-            sessionRestored = true
-            checkIfShouldDisplayBar()
+    func didRestoreSession() {
+        guard !sessionRestored else { return }
+        if let tabManager = self.tabManager,
+            let firstTab = tabManager.selectedTab {
+            observeURLForFirstTab(firstTab: firstTab)
+        } else {
+            firstTabLoaded = true
         }
+
+        sessionRestored = true
+        checkIfShouldDisplayBar()
```

Comment:
Cleanup item. Stop using a notification. The clipboardbar is directly accessible from BVC and it requires a single function call the replace this notification code which is in a few places.